ios/iosApp/UI/BaseViewModelViewController.swift

```diff
@@ -0,0 +1,29 @@
+import TrikotFrameworkName
+import UIKit
+
+class BaseViewModelViewController<V: BaseViewModelView<VM>, VMC: AbstractViewModelController, VM: ViewModel>: UIViewController {
+    var mainView: V {
+        view as! V
+    }
+
+    let viewModelController: VMC
+
+    init(viewModelController: VMC) {
+        self.viewModelController = viewModelController
+        super.init(nibName: nil, bundle: nil)
+    }
```

Comment:
```
    let viewModelController: VMC
    private let viewModel: VM
    
    init(viewModelController: VMC) {
        self.viewModelController = viewModelController
        // swiftlint:disable force_cast
        self.viewModel = viewModelController.viewModel as! VM
        // swiftlint:enable force_cast
        super.init(nibName: nil, bundle: nil)
        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: UIBarButtonItem.Style.plain, target: nil, action: nil)
    }
```

Reply:
What do you think about that way ? UIViewController can grad the title of the viewModel and having all the casting done in the init feels cleaner

Reply:
Why keeping a reference to the viewModel in the viewController ? Lets discuss that afterward.

Reply:
Really depends on where the info for the navigationBar would be. 

Maybe a ViewModelController could have a "navigationViewModel". Not sure if it is too iOS specific. Food for though, let's keep this for a future improvement when we feel we better nailed the pattern

Reply:
I really feel like the navigation bar should fall under the responsibility of the ViewModelController and I don't think that the viewModelController should keep a reference to the view model either. The title property should probably be in the VMC instead of the root VM if we want to keep that separation of concerns clean.

Reply:
That would be a very "iOS" specific way of supporting it. To be more platform agnostic, maybe having the VMC expose 2 ViewModels would make more sens. 

viewModel
navigationBarModel

Reply:
Not exactly sure why it is an "iOS" concept, to my knowledge the `ActionBar` on Android  is directly managed by the Activity and contains a title, a subtitle and a navigation button. Sounds very similar to me.

Reply:
What i mean by that is: On android, the actionBar is a "true" custom view that you customize the way you like. That is why I think a "ActionBarModel/NavigationBarModel" exposed by the ViewModelController would make more sens than having the ViewModelController expose properties directly. IMO ViewModelController should only take care of navigation and other orchestrations. Nothing related to UI

Reply:
The navigation bar is kind of both navigation related and a bit UI related. I don't see any problem exposing the behavioural part of a navigation bar in the view model controller itself, leaving the UI part the the view. The same way `UIBarButton` does not inherit from `UIView` but only describe the bar content and actions. At some point views will be created and added to the view hierarchy but at that point the controller is no longer involved.

Reply:
i think we are saying almost the same thing. I am only adding that this could be encapsulated in a NavigationBarModel to better separate the concern. No ?

Reply:
I am just unsure wether or not those nav bar related properties should inherit from a view model in any way 🤔. I would have put a plain string for the title and not really sure for nav buttons.

Reply:
Good point. That might just be a NavigationModel. 
Wondering if we could use `bind` with `keypath`to bind let's say navigationItem.title 

Would be worth having a publisher in that case

Reply:
And doing it with publishers would allow full databinding on Android